scipy/stats/tests/test_continuous_basic.py

```diff
@@ -106,13 +107,13 @@ def test_cont_basic(distname, arg):
     except TypeError:
         distfn = distname
         distname = 'rv_histogram_instance'
-    np.random.seed(765456)
+    rng = np.random.RandomState(765456)
     sn = 500
     with npt.suppress_warnings() as sup:
         # frechet_l and frechet_r are deprecated, so all their
         # methods generate DeprecationWarnings.
         sup.filter(category=DeprecationWarning, message=".*frechet_")
-        rvs = distfn.rvs(size=sn, *arg)
+        rvs = distfn.rvs(size=sn, *arg, random_state=rng)
```

Comment:
Are these changes (`random_state = ...` instead of `np.random.seed`) necessary for the tests to pass or is it a cleanup?

If it's a cleanup, the cleanup's good but maybe better separate them to their own PR.  If they are needed, then why --- np.random.seed was OK before and is no longer OK with this PR?

Reply:
When the distributions are first created on import the following is True:
```
from scipy.stats import norm
assert norm._random_state is np.random.mtrand._rand
```
i.e. the `_random_state` attribute is the numpy global RandomState singleton object. Thus setting the random seed via `np.random.seed` affects the random number stream from `norm.rvs`. However, as I was making these changes I found that various tests were failing, those that relied on a specific output from random number generation. I then put the assert given above into the failing tests, and the assert failed. I therefore suspect that in some cases (possibly a pytest side effect), the numpy RandomState singleton is not the same object as that contained in all the distributions. I don't know why that is. That being the case, using `np.random.seed` would have no effect on the stream of `distfn.rvs`.
I made changes like the line this comment is attached to on the failing tests, and I also decided to make similar changes to the test files they contained. I have seen some of the same stats tests being flaky from time to time, this may be the cause.